course4/homework1/vector2d.h

```diff
@@ -1,11 +1,41 @@
 #pragma once
 
+#include <cmath>
 #include <ostream>
 
 struct Vector2d
 {
 public:
-    Vector2d(int x, int y) : mX(x), mY(y) {}
+    // named ctors
+    static Vector2d fromCarthesian(const double x, const double y) {
+        return { x, y };
+    }
+    static Vector2d fromPolar(const double rho, const double phi) {
+        return { rho * cos(phi), rho * sin(phi) };
+    }
+    // ctor
+    Vector2d(double x, double y) : mX(x), mY(y) {}
```

Comment:
It may be a good idea to make the constructor private, to avoid confusion and get consistent API